test/unit/storage/FileResourceStore.test.ts

```diff
@@ -0,0 +1,572 @@
+import { ConflictHttpError } from '../../../src/util/errors/ConflictHttpError';
+import { DataFactory } from 'n3';
+import { FileResourceStore } from '../../../src/storage/FileResourceStore';
+import { InteractionController } from '../../../src/util/InteractionController';
+import { join as joinPath } from 'path';
+import { MethodNotAllowedHttpError } from '../../../src/util/errors/MethodNotAllowedHttpError';
+import { NotFoundHttpError } from '../../../src/util/errors/NotFoundHttpError';
+import { Readable } from 'stream';
+import { RepresentationMetadata } from '../../../src/ldp/representation/RepresentationMetadata';
+import streamifyArray from 'streamify-array';
+import { UnsupportedMediaTypeHttpError } from '../../../src/util/errors/UnsupportedMediaTypeHttpError';
+import { CONTENT_TYPE_QUADS, DATA_TYPE_BINARY, DATA_TYPE_QUAD } from '../../../src/util/ContentTypes';
+import fs, { promises as fsPromises, Stats } from 'fs';
+import { LDP, RDF, STAT, TERMS, XML } from '../../../src/util/Prefixes';
+import { LINK_TYPE_LDP_BC, LINK_TYPE_LDPR } from '../../../src/util/LinkTypes';
+import { literal, namedNode, quad as quadRDF, triple } from '@rdfjs/data-model';
+
+const base = 'http://test.com/';
+const root = '/Users/default/home/public/';
+
+fsPromises.rmdir = jest.fn();
+fsPromises.lstat = jest.fn();
+fsPromises.readdir = jest.fn();
+fsPromises.mkdir = jest.fn();
+fsPromises.unlink = jest.fn();
+fsPromises.access = jest.fn();
+
+describe('A FileResourceStore', (): void => {
+  let store: FileResourceStore;
+  let representation: BinaryRepresentation;
+  let readableMock: Readable;
+  let stats: Stats;
+  let stats1: Stats;
+  let stats2: Stats;
+  const rawData = 'lorem ipsum dolor sit amet consectetur adipiscing';
+  const quad = triple(
+    namedNode('http://test.com/s'),
+    namedNode('http://test.com/p'),
+    namedNode('http://test.com/o'),
+  );
+
+  fs.createReadStream = jest.fn();
+
+  beforeEach(async(): Promise<void> => {
+    jest.clearAllMocks();
+
+    store = new FileResourceStore(base, root, new InteractionController());
+
+    representation = {
+      data: streamifyArray([ rawData ]),
+      dataType: DATA_TYPE_BINARY,
+      metadata: { raw: [], linkRel: { type: new Set() }} as unknown as RepresentationMetadata,
+    };
+
+    stats = {
+      isDirectory(): boolean {
+        return false;
+      },
+      isFile(): boolean {
+        return false;
+      },
+      size: 0,
+      mtime: new Date(),
+    } as unknown as Stats;
+    stats1 = {
+      isDirectory(): boolean {
+        return false;
+      },
+      isFile(): boolean {
+        return false;
+      },
+      size: 0,
+      mtime: new Date(),
+    } as unknown as Stats;
+    stats2 = {
+      isDirectory(): boolean {
+        return false;
+      },
+      isFile(): boolean {
+        return false;
+      },
+      size: 0,
+      mtime: new Date(),
+    } as unknown as Stats;
```

Comment:
Don't we have possibilities of automocking? I also remember doing stuff like this https://github.com/RubenVerborgh/solid-server-ts/blob/1af325bbce3838c818eef0bd8b45955b548e9c67/test/unit/http/ResourceStoreRequestHandler.ts#L23-L25